api/src/main/java/org/pmiops/workbench/api/ClusterController.java

```diff
@@ -0,0 +1,130 @@
+    firecloudClusterRequest.setLabels(allOfUsClusterRequest.getLabels());
+    firecloudClusterRequest.setJupyterExtensionUri(allOfUsClusterRequest.getJupyterExtensionUri());
+
+    return firecloudClusterRequest;
+  }
+
+  private String convertClusterName(String workspaceId) {
+      String clusterName = workspaceId + this.userProvider.get().getUserId();
+      clusterName = clusterName.toLowerCase();
+      return clusterName;
+  }
+
+  @Autowired
+  ClusterController(NotebooksService notebooksService,
+      Provider<User> userProvider) {
+    this.notebooksService = notebooksService;
+    this.userProvider = userProvider;
+  }
+
+  public ResponseEntity<Cluster> createCluster(String workspaceNamespace,
+      String workspaceId,
+      ClusterRequest clusterRequest) {
+    Cluster createdCluster;
+
+    String clusterName = this.convertClusterName(workspaceId);
+    try {
+      // TODO: Replace with real workspaceNamespace/billing-project
+      createdCluster = toAllOfUsCluster(this.notebooksService.createCluster("broad-dsde-dev", clusterName, toFirecloudClusterRequest(clusterRequest)));
+    } catch (ApiException e) {
+      // TODO: Actually handle errors reasonably
+      throw new RuntimeException(e);
+    }
+    return ResponseEntity.ok(createdCluster);
+  }
+
+
+  public ResponseEntity<EmptyResponse> deleteCluster(String workspaceNamespace,
+      String workspaceId) {
+    String clusterName = this.convertClusterName(workspaceId);
+    try {
+      // TODO: Replace with real workspaceNamespace/billing-project
+      this.notebooksService.deleteCluster("broad-dsde-dev", clusterName);
+    } catch (ApiException e) {
+      // TODO: Actually handle errors reasonably
+      throw new RuntimeException(e);
+    }
+    EmptyResponse e = new EmptyResponse();
+    return ResponseEntity.ok(e);
+  }
+
+
+  public ResponseEntity<Cluster> getCluster(String workspaceNamespace,
+      String workspaceId) {
+    String clusterName = this.convertClusterName(workspaceId);
+    Cluster cluster;
+    try {
+      // TODO: Replace with real workspaceNamespace/billing-project
+      cluster = toAllOfUsCluster(this.notebooksService.getCluster("broad-dsde-dev", clusterName));
+    } catch(ApiException e) {
+      // TODO: Actually handle errors reasonably
+      throw new RuntimeException(e);
+    }
+    return ResponseEntity.ok(cluster);
+  }
+
+
+  public ResponseEntity<List<Cluster>> listClusters(String labels) {
+    List<Cluster> newClusters = new ArrayList<Cluster>();
+    List<org.pmiops.workbench.notebooks.model.Cluster> oldClusters;
+    try {
+      oldClusters = this.notebooksService.listClusters(labels);
+    } catch(ApiException e) {
+      // TODO: Actually handle errors reasonably
+      throw new RuntimeException(e);
+    }
+    for (org.pmiops.workbench.notebooks.model.Cluster c : oldClusters) {
+      newClusters.add(toAllOfUsCluster(c));
+    }
+    return ResponseEntity.ok(newClusters);
```

Comment:
Another potential API change... for cohorts we have wrapped the list of cohorts returned as an "items" attribute of the response, e.g. 

https://github.com/all-of-us/workbench/blob/master/api/src/main/java/org/pmiops/workbench/api/CohortController.java#L125

https://github.com/all-of-us/workbench/blob/master/api/src/main/resources/workbench.yaml#L513

That gives us the flexibility to add other metadata to the response in future. 

We should probably do the same for clusters.